mitodl/mit-open
# cache learning resources search api view

### What are the relevant tickets?

Closes #https://github.com/mitodl/hq/issues/5162

### Description (What does it do?)
This PR uses the django cache to cache the learning resources search api endpoint in addition to clearing out the caches at the end of each backpopulate task.

### How can this be tested?
1. checkout this branch
2. before doing anything, open up the django shell and view the current cache keys (should be empty)
```
from django.core.cache import caches
cache = caches['redis']
print(cache.keys('views.decorators.cache.cache_page.search.*'))
```
3. in learning_resources_search/views.py add some visible logging such as ```print("TESTING 123", "\n" * 20)``` to the get() method in LearningResourcesSearchView - this is so that we can visibly observe if the page is being served from the cache or not.
4. go to a search page and start browsing around - observe that on the first load of any page + url params the logging is visible but on subsequent loads it is served from the cache and does not output any print statements.
5. go back to the django shell and get the cache keys and observe there are entries for the search pages:
```
from django.core.cache import caches
cache = caches['redis']
print(cache.keys('views.decorators.cache.cache_page.search.*'))

```


### Additional Context
This is the first step into caching all anonymous views. Once we deploy this and validate, we can apply this approach to the remaining endpoints
